tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +55,111 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
+        /// <param name="items">Multipart form data items</param>
+        /// <returns>Task with the response</returns>
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + Boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+                var boundaryBytes = _boundarySeparatorInBytes ??= Encoding.ASCII.GetBytes(BoundarySeparator);
+                var trailerBytes = _boundaryTrailerInBytes ??= Encoding.ASCII.GetBytes(BoundaryTrailer);
+
+                // Write each MultipartFormItem
+                var hasWritten = false;
+                for (var i = 0; i < items.Length; i++)
+                {
+                    var item = items[i];
+                    byte[] headerBytes = null;
+
+                    // Check name is not null (required)
+                    if (item.Name is null)
+                    {
+                        Log.Warning("Error encoding multipart form item name is null. Ignoring item");
+                        continue;
+                    }
+
+                    // Ignore the item if the name contains ' or "
+                    if (item.Name.IndexOf("\"", StringComparison.Ordinal) != -1 || item.Name.IndexOf("'", StringComparison.Ordinal) != -1)
+                    {
+                        Log.Warning("Error encoding multipart form item name: {Name}. Ignoring item.", item.Name);
+                        continue;
+                    }
+
+                    // Do the same checks for FileName if not null
+                    if (item.FileName is not null)
+                    {
+                        // Ignore the item if the name contains ' or "
+                        if (item.FileName.IndexOf("\"", StringComparison.Ordinal) != -1 || item.FileName.IndexOf("'", StringComparison.Ordinal) != -1)
+                        {
+                            Log.Warning("Error encoding multipart form item filename: {FileName}. Ignoring item.", item.FileName);
+                            continue;
+                        }
+
+                        headerBytes = Encoding.ASCII.GetBytes(
+                            $"Content-Type: {item.ContentType}\r\nContent-Disposition: form-data; name=\"{item.Name}\"; filename=\"{item.FileName}\"\r\n\r\n");
+                    }
+
+                    headerBytes ??= Encoding.ASCII.GetBytes(
+                        $"Content-Type: {item.ContentType}\r\nContent-Disposition: form-data; name=\"{item.Name}\"\r\n\r\n");
+
+                    if (i == 0)
```

Comment:
I'm not sure if `i == 0` is the right check, because what happens if there was an error with the first item? What if `hasWritten` was changed to an int that increments at the end of each loop and the comparison is updated to `hasWritten == 0`?